src/plugin/featureaction/featuresoundaction.js

```diff
@@ -0,0 +1,183 @@
+goog.provide('plugin.im.action.feature.SoundAction');
+goog.require('os.feature');
+goog.require('os.im.action.AbstractImportAction');
+goog.require('os.implements');
+goog.require('os.legend');
+goog.require('os.legend.ILegendRenderer');
+goog.require('os.object');
+goog.require('os.xml');
+
+/**
+ * Tag names used for XML persistence.
+ * @enum {string}
+ */
+plugin.im.action.feature.SoundActionTagName = {
+  SOUND: 'sound',
+  DELAY: 'playDelay'
+};
+
+/**
+ * Import action that sets the sound for a {@link ol.Feature}.
+ * @extends {os.im.action.AbstractImportAction<ol.Feature>}
+ * @implements {os.legend.ILegendRenderer}
+ * @constructor
+ */
+plugin.im.action.feature.SoundAction = function() {
+  plugin.im.action.feature.SoundAction.base(this, 'constructor');
+
+  this.id = plugin.im.action.feature.SoundAction.ID;
+  this.label = plugin.im.action.feature.SoundAction.LABEL;
+  this.xmlType = plugin.im.action.feature.SoundAction.ID;
+  this.configUI = plugin.im.action.feature.SoundAction.CONFIG_UI;
+
+  /**
+   * Unique identifier for this action.
+   * @type {number}
+   * @protected
+   */
+  this.uid = goog.getUid(this);
+
+  /**
+   * The feature sound config.
+   * @type {!Object}
+   */
+  this.soundConfig = /** @type {!Object} */ (os.object.unsafeClone(
+      plugin.im.action.feature.SoundAction.DEFAULT_CONFIG));
+
+  /**
+   * User defined time between sound notifications in seconds.
+   * @type {number}
+   */
+  this.delay = this.soundConfig['playDelay'] * 1000;
+
+  this.refreshTimer_ = os.debounce(this.onRefreshTimer_, this.delay, true);
+};
+goog.inherits(plugin.im.action.feature.SoundAction,
+    os.im.action.AbstractImportAction);
+
+os.implements(plugin.im.action.feature.SoundAction,
```

Comment:
Sound actions will not be shown in the legend, so this class should not implement `ILegendRenderer`.